src/main/java/com/rackspace/ceres/app/config/CacheConfig.java

```diff
@@ -16,11 +16,39 @@
 
 package com.rackspace.ceres.app.config;
 
-import org.springframework.cache.annotation.EnableCaching;
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.rackspace.ceres.app.model.SeriesSetCacheKey;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.cache.CaffeineCacheMetrics;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
-@EnableCaching
 public class CacheConfig {
 
+  private final MeterRegistry meterRegistry;
+  private final AppProperties appProperties;
+
+  @Autowired
+  public CacheConfig(MeterRegistry meterRegistry,
+                     AppProperties appProperties) {
+    this.meterRegistry = meterRegistry;
+    this.appProperties = appProperties;
+  }
+
+  @Bean
+  public AsyncCache<SeriesSetCacheKey,Boolean/*exists*/> seriesSetExistenceCache() {
+    final AsyncCache<SeriesSetCacheKey, Boolean> cache = Caffeine
```

Comment:
Why caffeine and not ehcache?  something to do with the async stuff?

Reply:
Yep, from PR description:

> In-memory is specifically using Caffeine directly rather than Spring Cache, since Spring Cache (and JCache in general) does not yet support async/reactive cache-miss loading.

Reply:
We are using caffeine or redis cache?